src/zenml/model_deployers/base_model_deployer.py

```diff
@@ -226,9 +400,43 @@ def stop_model_server(
                 set to 0, the method will return immediately after
                 deprovisioning the service, without waiting for it to stop.
             force: if True, force the service to stop.
+
+        Raises:
+            RuntimeError: if the model server is not found.
         """
+        client = Client()
+        try:
+            service = self.find_model_server(service_uuid=uuid)[0]
+            updated_service = self.perform_stop_model(service, timeout, force)
+            client.update_service(
+                id=updated_service.uuid,
+                admin_state=updated_service.admin_state,
+                status=updated_service.status.dict(),
+                endpoint=updated_service.endpoint.dict()
+                if updated_service.endpoint
+                else None,
+            )
+        except Exception as e:
+            raise RuntimeError(
+                f"Failed to stop model server with UUID {uuid}: {e}"
+            ) from e
 
     @abstractmethod
+    def perform_start_model(
+        self,
+        service: BaseService,
+        timeout: int = DEFAULT_DEPLOYMENT_START_STOP_TIMEOUT,
+    ) -> BaseService:
+        """Abstract method to start a model server.
+
+        Args:
+            service: The service to start.
```

Comment:
see down here you talk about a service, but the method name talks about a model, and then in the docstring you talk about a model server..